### PR TITLE
Implement PlanNode::addDetails for all plan nodes

### PR DESCRIPTION
**Pull request by mbasmanova**

### CODE REVIEW

File: velox/exec/tests/PlanNodeToStringTest.cpp [hunk 1]
```diff
@@ -13,67 +13,336 @@
...
-    auto vector = makeRowVector(
-        {makeFlatVector<int32_t>(100, [](auto row) { return row; }),
-         makeFlatVector<int32_t>(100, [](auto row) { return row + 1; })});
-    vectors_.push_back(vector);
+    functions::prestosql::registerAllScalarFunctions();
+    parse::registerTypeResolver();
+    data_ = makeRowVector({
+        makeFlatVector<int16_t>({0, 1, 2, 3, 4}),
+        makeFlatVector<int32_t>({0, 1, 2, 3, 4}),
+        makeFlatVector<int64_t>({0, 1, 2, 3, 4}),
```
Comment (majetideepak) [inline velox/exec/tests/PlanNodeToStringTest.cpp:33]: The third row (`c2`) is not used?

File: velox/exec/tests/PlanNodeToStringTest.cpp [hunk 2]
```diff
@@ -13,67 +13,336 @@
...
+      plan_->toString(true, false));
 }
 
 TEST_F(PlanNodeToStringTest, recursiveAndDetailed) {
-  auto output = plan_->toString(true, true);
   ASSERT_EQ(
       "->project[expressions: (out3:BIGINT, plus(cast ROW[\"out1\"] as BIGINT,10)), ]\n"
       "  ->filter[expression: lt(mod(cast ROW[\"out1\"] as BIGINT,10),8)]\n"
-      "    ->project[expressions: (out1:INTEGER, ROW[\"c0\"]), (out2:BIGINT, plus(mod(cast ROW[\"c0\"] as BIGINT,100),mod(cast ROW[\"c1\"] as BIGINT,50))), ]\n"
+      "    ->project[expressions: (out1:SMALLINT, ROW[\"c0\"]), (out2:BIGINT, plus(mod(cast ROW[\"c0\"] as BIGINT,100),mod(cast ROW[\"c1\"] as BIGINT,50))), ]\n"
```
Comment (majetideepak) [inline velox/exec/tests/PlanNodeToStringTest.cpp:73]: missing spaces after the separator `,`. Not related to this change, but something to fix later.

File: velox/core/PlanNode.cpp [hunk 3]
```diff
@@ -99,21 +99,67 @@ AggregationNode::AggregationNode(
   }
 }
 
+namespace {
+void addKeys(
+    std::stringstream& stream,
+    const std::vector<std::shared_ptr<const FieldAccessTypedExpr>>& keys) {
+  for (auto i = 0; i < keys.size(); ++i) {
```
[the remark this answers is not in the document]
Comment (mbasmanova) [reply velox/core/PlanNode.cpp:106]: Good to know. In this particular code, I think we care more about readability though.

File: velox/exec/tests/PlanNodeToStringTest.cpp [hunk 4]
```diff
@@ -13,67 +13,336 @@
...
+                     .project({"c0 as u_c0", "c1 as u_c1"})
+                     .planNode(),
+                 "t_c1 > u_c1",
+                 {"t_c0", "t_c1", "u_c1"},
+                 core::JoinType::kLeft)
+             .planNode();
+
+  ASSERT_EQ("->hash join\n", plan->toString());
+  ASSERT_EQ(
+      "->hash join[LEFT t_c0=u_c0, filter: gt(ROW[\"t_c1\"],ROW[\"u_c1\"])]\n",
```
[the remark this answers is not in the document]
Comment (mbasmanova) [reply velox/exec/tests/PlanNodeToStringTest.cpp:139]: Agree. I'm reusing ITypedExpr::toString() here. I'll look into changing that as a separate PR.

File: velox/exec/tests/PlanNodeToStringTest.cpp (hunk 1 above)
Comment (mbasmanova) [reply velox/exec/tests/PlanNodeToStringTest.cpp:33]: Used in TEST_F(PlanNodeToStringTest, aggregation)

```
  auto plan = PlanBuilder()
                  .values({data_})
                  .partialAggregation(
                      {}, {"sum(c0) AS a", "avg(c1) AS b", "min(c2) AS c"})
```

**Comment by mbasmanova** [comment]
@majetideepak Thank you for review, Deepak.

**Review by aditi-pandit — COMMENTED** [review]
Thanks for this change Masha !

**Pull request closed without merging**